### PR TITLE
fix the regs conflicts within the `addResolution` for LoongArch64 and RISC-V.

### PR DESCRIPTION
**Pull request by shushanhf**
This PR is part of the issue https://github.com/dotnet/runtime/issues/69705 to amend the LA's port.

 The LoongArch64 is very different with the AArch64 and AMD64.
```
// For AArch64 and AMD64:                       |  // For LoongArch64
// cmp $reg1, $reg2  <--just compare.           |  the compare and branch can be written in one instruction.
// branch.condition  <--the condition is here.  |  so the comparing registers should be alive for branch.
```
If insert move before the branch just liking AArch64 or AMD64, there maybe conflicts.

This should be also valid for RISCV64. cc @clamp03

### CODE REVIEW

**Comment by shushanhf** [comment]
Hi, @jakobbotsch 
Could you please review this PR?
Thanks

**Comment by jakobbotsch** [comment]
@shushanhf This should be fixed in LSRA instead. It is what this TODO is about:

https://github.com/dotnet/runtime/blob/7486bdead70b884598c3288f1dfc232b2f077cc1/src/coreclr/jit/lsra.cpp#L8172-L8176

**Comment by shushanhf** [comment]
> @shushanhf This should be fixed in LSRA instead. It is what this TODO is about:
> 
> https://github.com/dotnet/runtime/blob/7486bdead70b884598c3288f1dfc232b2f077cc1/src/coreclr/jit/lsra.cpp#L8172-L8176

OK, Thanks.
I will amend it for LA.

File: src/coreclr/jit/lsra.cpp [hunk 1]
```diff
@@ -8155,23 +8155,29 @@ void LinearScan::handleOutgoingCriticalEdges(BasicBlock* block)
...
+            else if (op->IsLocal())
             {
                 GenTreeLclVarCommon* lcl = op->AsLclVarCommon();
                 terminatorNodeLclVarDsc  = &compiler->lvaTable[lcl->GetLclNum()];
             }
 
+            if (!lastNode->gtGetOp2()->isContainedIntOrIImmed() && (op != lastNode->gtGetOp2()))
+            {
+                op = lastNode->gtGetOp2();
+                goto Consume_Op2;
```
[the remark this answers is not in the document]
Comment (jakobbotsch) [reply src/coreclr/jit/lsra.cpp:8176]: Make sure to check for `GT_JCMP` and `GT_JTEST` before you access op2 (like below), and after you can remove the assert below.

File: src/coreclr/jit/lsra.cpp [hunk 2]
```diff
@@ -8162,16 +8162,31 @@ void LinearScan::handleOutgoingCriticalEdges(BasicBlock* block)
...
+                }
+                else if (op->IsLocal())
+                {
+                    GenTreeLclVarCommon* lcl = op->AsLclVarCommon();
+                    terminatorNodeLclVarDsc  = &compiler->lvaTable[lcl->GetLclNum()];
+                }
+            }
+#else
             assert(!lastNode->OperIs(GT_JCMP, GT_JTEST) || lastNode->gtGetOp2()->isContained());
 #endif
```
[the remark this answers is not in the document]
Comment (shushanhf) [reply src/coreclr/jit/lsra.cpp]: OK, thanks.

**Review by clamp03 — APPROVED** [review]
Thank you!!!

**Review by clamp03 — APPROVED** [review]
Thank you!!!

**Review by clamp03 — APPROVED** [review]
Thank you!!!

**Review by clamp03 — APPROVED** [review]
Thank you!!!

**Review by clamp03 — APPROVED** [review]
Thank you!!!

File: src/coreclr/jit/lsra.cpp [hunk 3]
```diff
@@ -8162,18 +8162,28 @@ void LinearScan::handleOutgoingCriticalEdges(BasicBlock* block)
...
+
+                if (op->OperIs(GT_COPY))
+                {
+                    GenTree* srcOp = op->gtGetOp1();
+                    consumedRegs |= genRegMask(srcOp->GetRegNum());
+                }
+                else if (op->IsLocal())
+                {
+                    GenTreeLclVarCommon* lcl = op->AsLclVarCommon();
+                    terminatorNodeLclVarDsc  = &compiler->lvaTable[lcl->GetLclNum()];
```
Comment (shushanhf) [inline src/coreclr/jit/lsra.cpp:8184]: The `terminatorNodeLclVarDsc` maybe duplicate, is this OK ?
@jakobbotsch

Comment (shushanhf) [reply src/coreclr/jit/lsra.cpp:8184]: OK, thanks.
I will add it.

**Review by jakobbotsch — APPROVED** [review]

**Comment by jakobbotsch** [comment]
Test failure is #86233 that is known by build analysis.

**Pull request merged**